File: api/src/main/java/com/example/api/dto/ItemDtoWithBookings.java
package com.example.api.dto;

import io.swagger.v3.oas.annotations.media.Schema;

import java.util.List;

@Schema(description = "Информация о предмете и его бронировании")
public record ItemDtoWithBookings(@Schema(description = "Идентификатор предмета") Long id,
                                  @Schema(description = "Название предмета") String name,
                                  @Schema(description = "Описание предмета") String description,
                                  @Schema(description = "Доступен ли предмет для аренды") Boolean available,
                                  @Schema(description = "Короткая информация о последнем бронировании") BookingShortDto lastBooking,
                                  @Schema(description = "Короткая информация о следующем бронировании") BookingShortDto nextBooking,
                                  @Schema(description = "Список отзывов") List<CommentDto> comments) {
}

File: api/src/main/kotlin/com/example/api/dto/ItemDtoWithBookings.kt
package com.example.api.dto

import io.swagger.v3.oas.annotations.media.Schema

@Schema(description = "Информация о предмете и его бронировании")
data class ItemDtoWithBookings(
    @get:Schema(description = "Идентификатор предмета") val id: Long,
    @get:Schema(description = "Название предмета") val name: String,
    @get:Schema(description = "Описание предмета") val description: String,
    @get:Schema(description = "Доступен ли предмет для аренды") val available: Boolean,
    @get:Schema(description = "Короткая информация о последнем бронировании") val lastBooking: BookingShortDto?,
    @get:Schema(description = "Короткая информация о следующем бронировании") val nextBooking: BookingShortDto?,
    @get:Schema(description = "Список отзывов") val comments: List<CommentDto?>?
)
